### ISSUE TITLE
-Werror=logical-not-parentheses problem on linux

### ISSUE THREAD

**Issue by andk** [dsully/perl-crypt-openssl-x509#48]
Since August my Linux smoker has not been able to compile Crypt-OpenSSL-X509-1.806:

``` text
cc -c  -I/usr/include/openssl -I/usr/local/include/ssl -I/usr/local/ssl/include -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -Wall -Werror   -DVERSION=\"1.806\" -DXS_VERSION=\"1.806\" -fPIC "-I/home/sand/src/perl/repoperls/installed-perls/perl/v5.22.1/9980/lib/5.22.1/x86_64-linux-thread-multi/CORE"   X509.c
X509.xs: In function ‘hv_exts’:
X509.xs:184:12: error: logical not is only applied to the left hand side of comparison [-Werror=logical-not-parentheses]
   if ( ! c > 0 ) {
            ^
X509.xs: In function ‘XS_Crypt__OpenSSL__X509_extension’:
X509.xs:871:10: error: logical not is only applied to the left hand side of comparison [-Werror=logical-not-parentheses]
   if (!c > 0) {
          ^
cc1: all warnings being treated as errors
Makefile:347: recipe for target 'X509.o' failed
make: *** [X509.o] Error 1
(/usr/bin/make exited with 512)
```

HTH&&Thanks,

**Comment by PhilippSoehnlein** [comment]
Same problem on Mac OS X 10.11 (El Capitan).
Setting the parenthesis the right way in X509.xs and running `perl Makefile.PL && make && make install` afterwards seems to work.

**Comment by jonasbn** [comment]
This seems to be fixed in PR #43 release: 1.807 - https://metacpan.org/source/DANIEL/Crypt-OpenSSL-X509-1.807
